### DISCUSSION TITLE
API Rules using system fields

### DISCUSSION THREAD

**Discussion by synapse** [pocketbase/pocketbase#1030] [Q&A] [answered]
I was trying to let users change their profile only after they've verified their email address using this rule:
`@request.auth.id != "" && id = @request.auth.id && verified = true`

The response is:

``` json
{
  "code": 400,
  "message": "Something went wrong while processing your request.",
  "data": {
    "updateRule": {
      "code": "validation_invalid_rule",
      "message": "Invalid filter rule."
    }
  }
}
```

Any idea if this is actually possibile or am I not writing the rule correctly?

**Comment by ganigeorgiev** [comment, marked as answer]
This looks like a bug. I'll check it a little bit later, but it seems that the system auth collection fields are not allowed in the rules validator.

**Reply by ganigeorgiev** [reply]
@synapse 

I've submitted a fix in the `rc` branch and updated the form tests to verify that it now works with system auth fields.

I'll try to push a v0.8.0-rc3 release a little bit later.

**Reply by ganigeorgiev** [reply]
@synapse Just to let you know that the fix is now available in the [v0.8.0-rc3 release](https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0-rc3).

**Reply by synapse** [reply]
Awesome, thanks @ganigeorgiev
